string flag
string from_node
float64 timestamp
